solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java

```diff
@@ -279,12 +282,20 @@ public void init(PluginInfo info) {
     int soTimeout =
         getParameter(args, HttpClientUtil.PROP_SO_TIMEOUT, HttpClientUtil.DEFAULT_SO_TIMEOUT, sb);
 
+    int keystoreReloadIntervalSecs =
```

Comment:
I think this should be handled on the Http2SolrClient (and HttpSolrClient) itself, kind of like the keystore/truststore/etc are configured through sysProps by default. There are numerous usages of Solr clients throughout Solr, and presumably, all of them would need to offer this. (Though I know most are short-lived).

Reply:
I'm fine either way. I didn't like much to "leak" the server-side system properties into the client, but it's true that it's what we doing with other settings, so maybe better to favor consistency.
I did explicitly disabled this for some of those short-lived clients to prevent them from starting and stopping the scanner.
Also, I was missing the setting for the update client, this new code you suggested is handling that. Added a test for it